packages/rancher-alerting-drivers/charts/templates/cluster-role.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if and (not .Values.sachet.enabled) (not .Values.prom2teams.enabled) -}}
+{{- fail "At least one Driver must be enabled to install the chart. " }}
+{{- end -}}
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ printf "%s-admin" (include "drivers.fullname" .) }}
+  labels: {{ include "drivers.labels" . | nindent 4 }}
+    rbac.authorization.k8s.io/aggregate-to-admin: "true"
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - configmaps
+    resourceNames: {{ include "drivers.configmapList" . | nindent 6 }}
```

Comment:
Now that I look at this, I'm actually slightly concerned that this list could return an empty string and we don't define any resourceNames and they get * on all configmaps in whatever namespace they are in.

Reply:
The case of empty string should never happen because of 2 reasons:

1. the app `rancher-alerting-drivers` will error out at the step of creating cluster roles during the installation if none of the sub-charts is enabled.
```
{{- if and (not .Values.sachet.enabled) (not .Values.prom2teams.enabled) -}}
{{- fail "At least one Driver must be enabled to install the chart. " }}
{{- end -}}
``` 

2. The template `drivers.configmapList` generates the list based on the full name of the enabled sub-charts, so the list will never be empty. 
```
{{/*
Get the list of configMaps to be managed
*/}}
{{- define "drivers.configmapList" -}}
{{- if .Values.sachet.enabled -}}
- {{ include "call-nested" (list . "sachet" "sachet.fullname") }}
{{- end }}
{{- if .Values.prom2teams.enabled -}}
- {{ include "call-nested" (list . "prom2teams" "prom2teams.fullname") }}
{{- end }}
{{- end }}
```